Tools/clinic/clinic.py

```diff
@@ -2180,7 +2205,7 @@ def print_block(
             *,
             core_includes: bool = False,
             limited_capi: bool,
-            header_includes: dict[str, str],
+            header_includes: dict[str, tuple[str, str]],
```

Comment:
Could you create a type alias for `tuple[str, str]` somewhere and use that in the type hint here? It could make it more readable, and you could reuse the type alias in your `sort_key()` function down below. Something like

```py
IncludeAndCondition = tuple[str, str]
```

Though I wonder if `dict[str, tuple[str, str]]` is really the best data structure to use here. Perhaps you could create an `Include` dataclass, e.g.:

```py
@dataclass
class Include:
    header_file: str
    reason: str
    condition: str | None
```

Reply:
Good idea. I added your proposed Include dataclass to my PR. You're right, it makes the code easier to understand.

Reply:
Thanks, I like it much more now!